test: created PyTest harness for SHACL  testing

Markus92 opened the pull request:
Initial pytest harness for the SHACLs

Right now I used parametrized tests for the core shapes. It'll be possible to add more tests, also tests that have to fail, xpass/xfail tests etc.

PatrickDekkerHealthRI commented on tests/test_shacl.py:
```diff
@@ -0,0 +1,23 @@
+    ],
+)
+def test_core_shapes(shacl, example):
+    shacl_graph = rf"Formalisation(shacl)/Core/PiecesShape/{shacl}.ttl"
```
Can you move the paths out of this function ? I prefer to have additional functions getTest() and getShacl().

Markus92 replied:
Done

PatrickDekkerHealthRI commented on tests/test_shacl.py:
```diff
@@ -0,0 +1,23 @@
+        ("Distribution", "example-distribution"),
+    ],
+)
+def test_core_shapes(shacl, example):
```
The name of the functions is not what you do.... i think the name shoudl be test_graph_conforms_shacl or something like that..

Markus92 replied:
This is the name that shows up in the rest report. I think `test_graph_conforms_shacl` might not be super descriptive, but open to suggestions.

PatrickDekkerHealthRI reviewed (changes requested):
Would be nice if there is bit more explanation on 

1. how to run the test locally
2.  How to add more test case. 
3. One example of failing validation would be nice:


I do have some code that check that for every shacl files at least an test file exists.. It matched the shacl on name so i was not really happy with that if can come up with a better way that would be nice...

kburger commented on tests/test_shacl.py:
I like that it's small and concise, and relatively easy to add new tests for newcomers. Patrick's comments are valid, but I'd be OK with refactoring later(ish).

Markus92 commented:
In the new version, I renamed the action to be more clear than just "ci" and made sure it runs on every PR (not just push, as that might skip it for PR's coming from different forks). I also added some basic documentation on how to run the test cases.

PatrickDekkerHealthRI reviewed (approved):
Good to go!

PatrickDekkerHealthRI reviewed (approved):
Ok
